Documentation/ApiOverview/ContentElements/Index.rst

```diff
@@ -110,21 +110,86 @@ your projects.
 possible without writing PHP code and can therefore also be done by
 TYPO3 integrators.
 
+..  _plugins:
 
+Plugins in TYPO3
+----------------
+
+A **plugin** in TYPO3 is a more complex module, typically providing dynamic
+or interactive functionality. Plugins are usually provided by extensions
+that introduce new features to the website.
+
+The data to be displayed is usually supplied by a special PHP class
+called a "controller". Depending on the technology used in the controller
+the plugin can be an Extbase plugin or a plain plugin.
 
-..  _plugins:
 ..  _plugins-extbase:
-..  todo: Introduce Extbase plugins
 
-What are plugins?
------------------
+Extbase plugins
+~~~~~~~~~~~~~~~
+
+For usage in the TYPO3 backend Extbase plugins are registered with utility
+functions of class :php:`\TYPO3\CMS\Extbase\Utility\ExtensionUtility` (not to
+be confused with :php:`\TYPO3\CMS\Core\Utility\ExtensionManagementUtility`).
+
+An Extbase plugin is configured for the frontend with
+:php:`ExtensionUtility::configurePlugin()` in file
+:file:`EXT:my_extension/ext_localconf.php`:
+
+..  literalinclude:: _Plugins/_ext_localconf_extbase_plugin.php
+    :caption: EXT:my_extension/ext_localconf.php
+
+By using `ExtensionUtility::PLUGIN_TYPE_PLUGIN` as fifth parameter is is also
+possible to add the plugin as a list type. See :ref:`plugins-list_type`.
+
+Method :php:`ExtensionUtility::configurePlugin()` also takes care of registering
+the plugin for frontend output in TypoScript using an object of type
+:ref:`EXTBASEPLUGIN <t3tsref:cobj-extbaseplugin>`.
+
+If it is desired that editors can insert the Extbase plugin like a content
+element into the page it also needs to be registered with
+:php:`ExtensionUtility::registerPlugin()` in the TCA Overrides, for example file
+:file:`EXT:my_extension/Configuration/TCA/Overrides/tt_content.php`:
+
+..  literalinclude:: _Plugins/_tt_content_extbase_plugin.php
+    :caption: EXT:my_extension/Configuration/TCA/Overrides/tt_content.php
+
+For a detailed discussion of Extbase plugins including examples for controllers
+see chapter :ref:`extbase`.
+
+..  _plugins-non-extbase:
+
+Plugins without Extbase
+~~~~~~~~~~~~~~~~~~~~~~~
+
+It is possible to create a plugin without using Extbase by creating a plain PHP
+class as a controller.
```

Comment:
Sorry. I have a problem with the "controller" wording. With "controller" I directly relate it to be part of the MVC pattern. It's a PHP class. And it's up to the coder to implement everything on his own or relate to the <out-dated> but still available AbstractPlugin class.

Reply:
An MVC pattern is not neccesarily bound to extbase. you can easily implement the controller pattern yourself.

Abstract plugin was deprecated with TYPO3 v12 and removed with TYPO3 v13 so we do not mention it here: https://docs.typo3.org/c/typo3/cms-core/main/en-us/Changelog/12.4/Deprecation-100639-DeprecateAbstractPlugin.html#deprecation-100639-deprecate-abstractplugin